### ISSUE TITLE
"ReportId" in Schema "DeviceAlertEvents" doesn't match "ReportId" in Schema "DeviceFileEvents"

### ISSUE THREAD

**Issue by banhao** [MicrosoftDocs/microsoft-365-docs#5688]
[Enter feedback here]
"ReportId" in Schema "DeviceAlertEvents" doesn't match "ReportId" in Schema "DeviceFileEvents"
Seems they are used to save different Events, but use the same name. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e9fd21fa-e367-78d4-bc86-80f2b75a64b8
* Version Independent ID: 97a42f10-ba7f-f2d4-6c5c-69e38a99f277
* Content: [DeviceAlertEvents table in the advanced hunting schema](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/advanced-hunting-devicealertevents-table?view=o365-worldwide)
* Content Source: [microsoft-365/security/defender-endpoint/advanced-hunting-devicealertevents-table.md](https://github.com/MicrosoftDocs/microsoft-365-docs/blob/public/microsoft-365/security/defender-endpoint/advanced-hunting-devicealertevents-table.md)
* Product: **m365-security**
* Technology: **mde**
* GitHub Login: @schmurky
* Microsoft Alias: **maccruz**

**Comment by schmurky** [comment]
Can you give me more information @banhao? what do you mean by they are used to save different events but use the same name? DeviceAlertEvents is only available in MDE (if that helps).

**Comment by banhao** [comment]
yes, I'm still using "https://securitycenter.windows.com/hunting"

I use the following query to get all the AlertEvents from a machine.
`DeviceAlertEvents | where DeviceName contains "machine_hostname"`

So I tried to use the following query to get the File Information that causes the alerts. But seems the "ReportId" in schema "DeviceAlertEvents" and "DeviceFileEvents" are totally different.

`let virusevents = DeviceAlertEvents | where DeviceName contains "machine_hostname";
DeviceFileEvents | join virusevents on ReportId`

**Comment by schmurky** [comment]
Hi @banhao sorry for the delay. ReportID always means the unique identifier of the event. In DeviceAlertEvents, it is the identifier of the event that triggered the alert, and it should exist in device tables.

Hopefully this doc can help: https://docs.microsoft.com/en-us/microsoft-365/security/defender/advanced-hunting-migrate-from-mde?view=o365-worldwide

Let me know what error you are encountering.
